tests/sentry/integrations/github/test_integration.py

```diff
@@ -585,9 +597,43 @@ def test_github_prevent_install_until_pending_deletion_is_complete(self):
             organization=self.organization, integration=integration
         ).exists()
 
+    def get_installation_helper(self):
+        with self.tasks():
+            self.assert_setup_flow()  # This somehow creates the integration
+
+        integration = Integration.objects.get(provider=self.provider.key)
+        installation = integration.get_installation(self.organization.id)
+        return installation
+
+    @responses.activate
+    def test_get_trees_for_org_handles_rate_limit_reached(self):
+        """Test that we will not hit Github's API more than once when we reach the API rate limit"""
+        installation = self.get_installation_helper()
+        # This will force reaching for xyz before foo
+        cache.set(
```

Comment:
This is a trick that allows me not to do mocking to change the order of how we iterate through the repositories.